traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -581,18 +581,28 @@ func ReadID(w http.ResponseWriter, r *http.Request) {
 		log.Warnf("Couldn't get config %v", e)
 	}
 	servers, _, userErr, sysErr, errCode, _ = getServers(r.Header, inf.Params, inf.Tx, inf.User, useIMS)
-
 	if len(servers) > 1 {
 		api.HandleDeprecatedErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("ID '%d' matched more than one server (%d total)", inf.IntParams["id"], len(servers)), &alternative)
 		return
 	}
-
 	deprecationAlerts := api.CreateDeprecationAlerts(&alternative)
 
 	// No need to bother converting if there's no data
 	if len(servers) < 1 {
 		api.WriteAlertsObj(w, r, http.StatusOK, deprecationAlerts, servers)
+		return
 	}
+	legacyServers := make([]tc.ServerNullableV11, 0, 1)
```

Comment:
as long as you're allocating memory ahead of time, you may as well set the capacity to what we know it needs to be; instead of 1 it'd be much more efficient to go ahead and allocate room for `len(servers)`.